### ISSUE TITLE
Quick fix for bin/test.sh in unit-test branch

### ISSUE THREAD

**Issue by cyrille-artho** [cyrille-artho/modbat#30]
Let's try to build the JAR file at the beginning of test.sh:

```
#!/bin/sh
CWD=`pwd`
cd build/classes/scala/test
jar cf build/modbat-examples.jar modbat/examples/
jar cf build/modbat-test.jar modbat/test
mv modbat-*.jar ../../..
cd ${CWD}

... # remaining script
```

**Comment by cyrille-artho** [comment]
This should really be done by gradle and is a q & d fix until we can figure out how to do this in gradle.

**Comment by cyrille-artho** [comment]
The current fix works to some extent for `bin/test.sh`, but the unit tests still fail because they also need a correct JAR file. Let's try to fix this in `build.gradle`:
https://docs.gradle.org/current/userguide/working_with_files.html#sec:creating_archives_example
Create two new targets in `build.gradle`: testJar and examplesJar. Add the necessary files with the right name for the output. This will hopefully fix both unit and system tests.

**Comment by cyrille-artho** [comment]
Compared to the examples shown on that page, the type of the archive should be `Jar`; the destinationDirectory should be `$buildDir`; `from` should contain `$buildDir/modbat/examples` and `$buildDir/modbat/tests`, respectively, for `modbat-examples.jar` and `modbat-test.jar`.
We can worry later about adding the source file.
The "test" task should depend on these two jar files; this can probably be done with
```
test {
  dependsOn ...
  dependsOn ...
}

unless the documentation I've found is wrong or outdated.

**Comment by cyrille-artho** [comment]
Fixed in f3e8b14. The JARs are still not quite right: `modbat-examples.jar` needs source files, and both JARs contain the union of the files that should be there. However, at least the tests run again successfully.
